### PR TITLE
Allow webmasters to add expand/collapse all links to individual forms.

### PR DESCRIPTION
**Pull request by richardbporter**
### Problem
All users cannot uncheck the "Display collapse/expand all details link" for Individual webforms:

image

This doesn't look good on the form I'm building nor does it add much value: https://diversity.uiowa.edu/form/report-a-problem

### Proposed solution
Uncheck the global webform setting for this config and let people add it back to individual webforms if they want.

# How to test
- Sync a site.
- Create a webform with more than 1 details element.
- See that the expand/collapse all links do not appear.
- See that you can add that in the form settings if you'd like.
- Review the SLEW of other webform settings changes that happened on export... 😬

### CODE REVIEW

**Comment by richardbporter** [comment]
Hm. I wonder if this will remove the functionality for existing webforms.

**Comment by richardbporter** [comment]
> Hm. I wonder if this will remove the functionality for existing webforms.

Yes it does. Is that ok? Or is updb needed?

**Comment by briand44** [comment]
Is this functionality needed for the webform you are working on?

**Comment by richardbporter** [comment]
> Is this functionality needed for the webform you are working on?

Expand/collapse all? Or details?

Either way, it seems like we should allow webmasters to toggle this. I think an update hook is unnecessary as this only affects webforms with two or more details elements. We could send an announcement saying this will no longer be the default so please update your forms as needed.

**Comment by joewhitsitt** [comment]
https://github.com/uiowa/uiowa/blob/main/config/tippie.uiowa.edu/webform.settings.yml needs to be updated

might want to test against https://github.com/uiowa/uiowa/blob/main/config/presidentialsearch.uiowa.edu/webform.webform.contact.yml to see if saving the form changes any config

**Comment by richardbporter** [comment]
> https://github.com/uiowa/uiowa/blob/main/config/tippie.uiowa.edu/webform.settings.yml needs to be updated
> 
> might want to test against https://github.com/uiowa/uiowa/blob/main/config/presidentialsearch.uiowa.edu/webform.webform.contact.yml to see if saving the form changes any config

Thanks. No config changes on presidentialsearch form.

**Comment by richardbporter** [comment]
Added change record to https://sitenow.uiowa.edu/news/2021/08/updates-august

File: config/default/webform.settings.yml [hunk 1]
```diff
@@ -1,38 +1,11 @@
...
-  default_submission_views: {  }
-  default_submission_views_replace:
-    global_routes:
-      - entity.webform_submission.collection
-      - entity.webform_submission.user
-    webform_routes:
-      - entity.webform.results_submissions
-      - entity.webform.user.drafts
-      - entity.webform.user.submissions
-    node_routes: {  }
```
Comment (richardbporter) [inline config/default/webform.settings.yml:28]: This was just moved below.

File: config/default/webform.settings.yml [hunk 2]
```diff
@@ -1,38 +1,11 @@
...
-  default_form_open_message: 'This form has not yet been opened to submissions.'
-  default_form_close_message: 'Sorry&hellip;This form is closed to new submissions.'
-  default_form_exception_message: 'Unable to display this webform. Please contact the site administrator.'
-  default_form_confidential_message: 'This form is confidential. You must <a href="[site:login-url]/logout?destination=[current-page:url:relative]">Log out</a> to submit it.'
-  default_form_access_denied_message: 'Please login to access this form.'
+  default_form_open_message: '<p>This form has not yet been opened to submissions.</p>'
+  default_form_close_message: '<p>Sorry…This form is closed to new submissions.</p>'
+  default_form_exception_message: '<p>Unable to display this webform. Please contact the site administrator.</p>'
+  default_form_confidential_message: '<p>This form is confidential. You must <a href="login-url]/logout?destination=[current-page:url:relative]">Log out</a> to submit it.</p>'
+  default_form_access_denied_message: '<p>Please login to access this form.</p>'
```
Comment (richardbporter) [inline config/default/webform.settings.yml:8]: Looks like these were just wrapped in markup.

**Comment by richardbporter** [comment]
I think a previous update hook changed some default webform settings that were never exported.

**Comment by bspeare** [comment]
Pulling to test.

**Review by bspeare — APPROVED** [review]
Tested on building with a new form and an existing form on Sandbox site.  Seems like a good fix!

**Comment by richardbporter** [comment]
Thanks!

**Pull request merged**